contracts/automation/andromeda-evaluation/src/contract.rs

```diff
@@ -0,0 +1,1045 @@
+pub fn reply(deps: Deps, _env: Env, reply: Reply) -> Result<Response, ContractError> {
+    // Load task balancer's address
+    let contract_addr = TASK_BALANCER_ADDRESS.load(deps.storage)?;
+    let app_contract = ADOContract::default().get_app_contract(deps.storage)?;
+    let contract_addr = contract_addr.get_address(deps.api, &deps.querier, app_contract.clone())?;
+    if let Some(app_address) = app_contract {
+        match reply.id {
+            1 => Ok(Response::new().add_submessage(SubMsg::new(CosmosMsg::Wasm(
+                WasmMsg::Execute {
+                    contract_addr,
+                    msg: to_binary(&RemoveProcess {
+                        process_address: app_address.into_string(),
+                    })?,
+                    funds: vec![],
+                },
+            )))),
+            _ => Err(ContractError::AccountNotFound {}),
+        }
+    } else {
+        Err(ContractError::AccountNotFound {})
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Evaluate {
+            user_value,
+            operation,
+        } => execute_evaluate(deps, env, info, user_value, operation),
+        ExecuteMsg::ChangeConditionAddress { address } => {
+            execute_change_condition_address(deps, env, info, address)
+        }
+        ExecuteMsg::ChangeQueryAddress { address } => {
+            execute_change_query_address(deps, env, info, address)
+        }
+    }
+}
+
+fn execute_change_query_address(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    address: AndrAddress,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    // Only the contract's owner can update the Execute ADO address
+    ensure!(
+        ADOContract::default().is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+    QUERY_ADO_ADDRESS.save(deps.storage, &address)?;
+    Ok(Response::new().add_attribute("action", "changed_query_ado_address"))
+}
+
+fn execute_change_condition_address(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    address: AndrAddress,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    // Only the contract's owner can update the Execute ADO address
+    ensure!(
+        ADOContract::default().is_owner_or_operator(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+    CONDITION_ADO_ADDRESS.save(deps.storage, &address)?;
+    Ok(Response::new().add_attribute("action", "changed_CONDITION_ADO_ADDRESS"))
+}
+
+fn execute_evaluate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    user_value: Uint128,
```

Comment:
Is user input here abuseable in any way? Can someone purposefully put an incorrect value here to trigger something that otherwise shouldn't be?

Reply:
The user_value is now stored during instantiation